src/GovUk.Education.ExploreEducationStatistics.Common/Model/Either.cs

```diff
@@ -281,8 +281,23 @@ public static class EitherTaskExtensions
             return await func(firstResult.Right);
         }
 
-        public static async Task<Either<TFailure1, TFaillure2>> OnFailureDo<TFailure1, TFaillure2>(
-            this Task<Either<TFailure1, TFaillure2>> task,
+        public static async Task<Either<Unit, TSuccess>> OnFailureVoid<TFailure, TSuccess>(
```

Comment:
You could add a test to EitherTests for this?

Reply:
Done!